src/chttpd/test/exunit/tracing_test.exs

```diff
@@ -0,0 +1,101 @@
+defmodule Couch.Test.OpenTracing do
+  use Couch.Test.ExUnit.Case
+  alias Couch.Test.Setup
+  alias Couch.Test.Setup.Step
+  alias Couch.Test.Utils
+  import Couch.DBTest, only: [retry_until: 1]
+
+  defp create_admin(user_name, password) do
+    hashed = String.to_charlist(:couch_passwords.hash_admin_password(password))
+    :config.set('admins', String.to_charlist(user_name), hashed, false)
+  end
```

Comment:
It seems like the `create_admin` function along with some of the other function below would be useful to have in the shared Couch library. Any particular reason to not put these functions into the `couch.ex` module?

Reply:
Few concerns:
- I tried that before [1] and [2]
  - the complain was: we are developing our own framework which people need to learn.
- Adding `create_admin` to `Couch` exibit a chicken and egg problem. `Couch` implements HTTP interface. We cannot use HTTP interface without adding admin first. In context of eunit tests we always cheated and created admin via config manipulation. 
- We could consider `create_admin` an exception and still add it to `Couch` inspite of it not being HTTP based interface. However it will introduce a confusion. People use `Couch` module from elixir integration tests and the function wouldn't work in the context of an integration test.


When I was writing this test I considered using `Step.User.new(:admin, roles: [:server_admin])`. However since pluguble adapters [1] didn't make it to the framework and [2] doesn't have fabric2 support, so I had to do it differently. I followed the advise of the committers who believe that DRY is bad for tests and they should be as self contained as possible.

---
[1]: [#2036](https://github.com/apache/couchdb/pull/2036) - plugable adaptors based approach
[2]: [#2039](https://github.com/apache/couchdb/pull/2039) - fabric only interface

Reply:
Ahhhh ok, thanks for the detailed explanation @iilyak 👍. It's outside the scope of this PR, but I wonder if it would make sense to add a tag like `@tag :with_admin` like we've done with dbs and what not [1], to inject an admin user context into the test and then we can abstract away whether the admin user was created through config or the http interface. As for framework vs anti DRY, initially with the Elixir suite we tried to avoid making a heavy framework, which I think is worthwhile, but at the same time there's common activities which should be easy to take care of, like creating dbs/docs/users/ddocs/etc.

[1] https://github.com/apache/couchdb/blob/master/test/elixir/test/security_validation_test.exs#L151-L153